packages/metal-dom/src/domNamed.js

```diff
@@ -521,6 +521,34 @@ export function parent(element, selector) {
 	return closest(element.parentNode, selector);
 }
 
+/**
+ * Inserts a node before first child of the parent. If child is a HTML string
+ * it will be converted to document fragment before prepending it to the parent.
+ * @param {!Element} parent The node to prepend to.
+ * @param {!(Element|NodeList|string)} child The thing who must be prepended
+ * @return {!Element} The prepended child
+ */
+export function prepend(parent, child) {
+	if (isString(child)) {
+		child = buildFragment(child);
+	}
+
+	if (!isNodeListLike(child) && !isDefAndNotNull(parent.firstChild)) {
+		parent.appendChild(child);
+	}
+
+	if (isNodeListLike(child)) {
+		const childArr = Array.prototype.slice.call(child);
+		for (let i = 0; i < childArr.length; i++) {
+			parent.appendChild(childArr[i]);
```

Comment:
If you add a node list to a parent that already has a child this won't work, this test will currently fail as an example.

```javascript
        it('should prepend a node list to parent element', function() {
            let parent = document.createElement('div');
            let child = document.createElement('div');

            dom.addClasses(child, 'child');
            dom.append(parent, child);

            let childFrag = dom.buildFragment(
                '<div class="myChild">el1</div><div class="myChild2">el2</div><div class="myChild3">el3</div>'
            );

            dom.prepend(parent, childFrag.childNodes);

            assert.strictEqual(4, parent.childNodes.length);
            assert.strictEqual('myChild', parent.childNodes[0].className);
            assert.strictEqual('myChild2', parent.childNodes[1].className);
            assert.strictEqual('myChild3', parent.childNodes[2].className);
            assert.strictEqual('child', parent.childNodes[3].className);
        });
```

Reply:
I've corrected this using insertBefore instead appendChild